### PR TITLE
add Node State Change

### PR DESCRIPTION
**Pull request by Charleslee522**
I think we need to add contents about changing node state.
Because this part is not documented in detail yet.

### CODE REVIEW

**Comment by cmcm2222** [comment]
Is it difference node state and consensus state?? 
then ISAAC is consensus state?

**Comment by Charleslee522** [comment]
> Is it difference node state and consensus state??
> then ISAAC is consensus state?

Yes, exactly.

**Comment by Charleslee522** [comment]
@cmcm2222 Thank you for your review. I've made the documentation a little richer.
So if you don't mind, would you review it again? :)

**Comment by cmcm2222** [comment]
why not for sure I will do.

File: isaac/how_to_sync_blocks_with_validators.md [hunk 1]
```diff
@@ -5,6 +5,24 @@
 1. `SYNC`      - The node is not ready for consensus because in sync.
 1. `WATCH`     - The node watches it's validators and try to sync. This node does not participate in the consensus.
 
+## Node State Change
+* `BOOTING` -> `CONSENSUS`
+The status changes from `BOOTING` to `CONSENSUS` when `SEBAK` runs and a sufficient number of validators are connected.
```
Comment (cmcm2222) [inline isaac/how_to_sync_blocks_with_validators.md]: when `SEBAK` runs and a sufficient number of validators are connected.

when node running and a number of validator nodes are connected.

File: isaac/how_to_sync_blocks_with_validators.md [hunk 2]
```diff
@@ -5,6 +5,24 @@
...
+* `CONSENSUS` -> `SYNC`
+The status changes from `CONSENSUS` to `SYNC` when its block height is lower than the agreed block height through voting.
```
Comment (cmcm2222) [inline isaac/how_to_sync_blocks_with_validators.md]: The node state changes from ~~~ when its block height lower than ~

File: isaac/how_to_sync_blocks_with_validators.md [hunk 3]
```diff
@@ -5,6 +5,24 @@
...
+   1. And when the height of this agreed ballot is different with `Last Confirmed Height + 1`.
+   1. The status changes from `CONSENSUS` to `SYNC`.
+* `SYNC` -> `CONSENSUS`
+The status changes from `SYNC` to `CONSENSUS` when,
+   * The node is in same height and round with the other validators.
+   * The node has all blocks, transactions and operations.
+   * All blocks the node has are valid.
+   * The node agrees on the proposer proposed block and save it.
+* `WATCH`
+The node set by Watcher is always `WATCH` state. That is, it does not change to another node state.
```
Comment (cmcm2222) [inline isaac/how_to_sync_blocks_with_validators.md]: set>> sets as the Watcher 
that is >> I thinks this is unnecessary.

File: isaac/how_to_sync_blocks_with_validators.md [hunk 4]
```diff
@@ -5,6 +5,24 @@
...
+* `BOOTING` -> `CONSENSUS`
+The status changes from `BOOTING` to `CONSENSUS` when `SEBAK` runs and a sufficient number of validators are connected.
+* `CONSENSUS` -> `SYNC`
+The status changes from `CONSENSUS` to `SYNC` when its block height is lower than the agreed block height through voting.
+   1. A node receives `ACCEPT` and `YES` ballots only.
+   1. When the number of ballots exceeds the threshold,
+   1. And when the height of this agreed ballot is different with `Last Confirmed Height + 1`.
+   1. The status changes from `CONSENSUS` to `SYNC`.
+* `SYNC` -> `CONSENSUS`
+The status changes from `SYNC` to `CONSENSUS` when,
```
Comment (cmcm2222) [inline isaac/how_to_sync_blocks_with_validators.md]: when >> in case of

File: isaac/how_to_sync_blocks_with_validators.md [hunk 5]
```diff
@@ -5,6 +5,24 @@
...
+   * The node is in same height and round with the other validators.
+   * The node has all blocks, transactions and operations.
+   * All blocks the node has are valid.
```
Comment (cmcm2222) [inline isaac/how_to_sync_blocks_with_validators.md]: all blocks that node has valid.

File: isaac/how_to_sync_blocks_with_validators.md [hunk 6]
```diff
@@ -5,6 +5,24 @@
...
+   * The node agrees on the proposer proposed block and save it.
```
Comment (cmcm2222) [inline isaac/how_to_sync_blocks_with_validators.md]: proposer's proposing block.

File: isaac/how_to_sync_blocks_with_validators.md (hunk 5 above)
Comment (Charleslee522) [reply isaac/how_to_sync_blocks_with_validators.md]: It means,
"All blocks (that the node has) are valid"

**Pull request merged**